From 4c7c3de6d286e784942c17b4c2a2aba3136db9d5 Mon Sep 17 00:00:00 2001
From: Boris Brezillon <boris.brezillon@collabora.com>
Date: Tue, 19 Sep 2023 17:23:22 +0200
Subject: [PATCH 61/64] panfrost: Prepare uniform buffers in
 panfrost_launch_xfb()

This is needed in csf_launch_xfb() too.
---
 src/gallium/drivers/panfrost/pan_cmdstream.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/gallium/drivers/panfrost/pan_cmdstream.c b/src/gallium/drivers/panfrost/pan_cmdstream.c
index c88afd6aa5f..c61ba4e647f 100644
--- a/src/gallium/drivers/panfrost/pan_cmdstream.c
+++ b/src/gallium/drivers/panfrost/pan_cmdstream.c
@@ -3575,11 +3575,6 @@ jm_launch_xfb(struct panfrost_batch *batch, const struct pipe_draw_info *info,
               mali_ptr attribs, mali_ptr attrib_bufs, UNUSED unsigned start,
               unsigned count)
 {
-   batch->uniform_buffers[PIPE_SHADER_VERTEX] =
-      panfrost_emit_const_buf(batch, PIPE_SHADER_VERTEX, NULL,
-                              &batch->push_uniforms[PIPE_SHADER_VERTEX],
-                              &batch->nr_push_uniforms[PIPE_SHADER_VERTEX]);
-
    struct panfrost_ptr t = pan_pool_alloc_desc(&batch->pool.base, COMPUTE_JOB);
 
 #if PAN_ARCH == 9
@@ -3709,6 +3704,11 @@ panfrost_launch_xfb(struct panfrost_batch *batch,
    batch->rsd[PIPE_SHADER_VERTEX] =
       panfrost_emit_compute_shader_meta(batch, PIPE_SHADER_VERTEX);
 
+   batch->uniform_buffers[PIPE_SHADER_VERTEX] =
+      panfrost_emit_const_buf(batch, PIPE_SHADER_VERTEX, NULL,
+                              &batch->push_uniforms[PIPE_SHADER_VERTEX],
+                              &batch->nr_push_uniforms[PIPE_SHADER_VERTEX]);
+
    JOBX(launch_xfb)(batch, info, attribs, attrib_bufs, start, count);
    batch->any_compute = true;
 
-- 
2.42.0

